hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -67,6 +67,13 @@ public class HoodieCompactionConfig extends HoodieConfig {
       .withDocumentation("When set to true, logcompaction service is triggered after each write. While being "
           + " simpler operationally, this adds extra latency on the write path.");
 
+  public static final ConfigProperty<String> DISABLE_ASYNC_COMPACT_FOR_SPARK_STREAMING = ConfigProperty
+      .key("hoodie.compact.disable.for.spark.streaming")
+      .defaultValue("false")
+      .withDocumentation("When set to true, compaction will not run inline nor asynchronously. Compaction can take up significant resources "
```

Comment:
some word smithing
```
By default for MOR table, async compaction is enabled with spark streaming sink. By setting this config to true, we can disable it and the expectation is that, users will schedule and execute compaction in a different process/job altogether. Some users may wish to run it separately to manage resources across table services and regular ingestion pipeline and so this could be preferred on such cases."
```